src/vscode/views/LispRepl.ts

```diff
@@ -125,46 +137,29 @@ export class LispRepl extends EventEmitter<ReplEvents> implements vscode.Webview
     private doEval(text: string) {
         if (text.trim().length != 0) {
             this.emit('eval', this.package, text)
-        } else {
-            this.addText('')
         }
     }
 
-    private getHtmlForView(webview: vscode.Webview): string {
-        const jsPath = vscode.Uri.file(path.join(this.ctx.extensionPath, 'resource', 'repl', 'view.js'))
-        const cssPath = vscode.Uri.file(path.join(this.ctx.extensionPath, 'resource', 'repl', 'view.css'))
+    private getWebviewContent(webview: vscode.Webview): string {
+        const jsPath = vscode.Uri.file(path.join(this.ctx.extensionPath, 'resources', 'repl', 'index.js'))
+        const cssPath = vscode.Uri.file(path.join(this.ctx.extensionPath, 'resources', 'repl', 'index.css'))
+        const scriptUri = webview.asWebviewUri(jsPath)
+        const stylesUri = webview.asWebviewUri(cssPath)
 
-        return `<!DOCTYPE html>
-                <html>
+        const nonce = getNonce()
+
+        return `
+            <!DOCTYPE html>
+            <html>
                 <head>
-                    <link rel="stylesheet" href="${webview.asWebviewUri(cssPath)}">
+                    <meta http-equiv="Content-Security-Policy" content="default-src 'none'; style-src ${webview.cspSource}; script-src 'nonce-${nonce}';">
+                    <link rel="stylesheet" type="text/css" href="${stylesUri}">
                 </head>
-
-                <body onfocus="setFocus()">
-                    <textarea id="repl-text" class="repl-text" readonly></textarea>
-                    <div class="repl-input-box">
-                        <div class="repl-input-text-box" id="repl-user-input-box">
-                            <div class="repl-input-label">
-                                Input >
-                            </div>
-                            <form id="repl-user-input-form" class="repl-input-form" action="">
-                                <input class="repl-input-text" id="repl-user-input" type="text" disabled>
-                            </form>
-                        </div>
-                        <div class="repl-input-text-box">
-                            <div class="repl-input-label" onclick="requestPackage()">
-                                <span id="repl-package">${this.package}</span>
-                                >
-                            </div>
-                            <form id="repl-input-form" class="repl-input-form" action="">
-                                <input class="repl-input-text" id="repl-input-text" type="text">
-                            </form>
-                        </div>
-                    </div>
-
-                    <script src="${webview.asWebviewUri(jsPath)}"></script>
+                <body>
+                    <div id="root"></div>
```

Comment:
It works to pass the initial values for package and the version as attributes to the root div and then passing those as props to `App` in `webview/index.tsx`. That might be simpler than trying to do it with message passing.